### PR TITLE
chore: rc-trigger move to dev depds

### PR DESCRIPTION
**Pull request by yoyo837**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at eeabb74</samp>

This pull request removes an unnecessary dependency and ensures consistent use of `rc-trigger` across components. This improves the development and reliability of the library.

### <samp>🤖 Generated by Copilot at eeabb74</samp>

* Remove unnecessary dependency on `rc-trigger` from `devDependencies` ([link](https://github.com/ant-design/ant-design/pull/41808/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L153))
* Enforce consistent version of `rc-trigger` across components using `resolutions` ([link](https://github.com/ant-design/ant-design/pull/41808/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R256))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -150,7 +150,6 @@
     "rc-tooltip": "~6.0.0",
     "rc-tree": "~5.7.0",
     "rc-tree-select": "~5.8.0",
-    "rc-trigger": "^5.3.4",
```
[the remark this answers is not in the document]
Comment (yoyo837) [reply package.json:153]: 我去, 我按 `rc-trigger'`搜索的, 没看到它.

[the remark this answers is not in the document]
Comment (afc163) [reply package.json:153]: 其实看看能不能改掉，比如从 rc-tooltip 透传出来。

antd 确实没必要直接依赖 rc-trigger

[the remark this answers is not in the document]
Comment (yoyo837) [reply package.json:153]: 我试试

**Comment by yoyo837** [comment]
@SocketSecurity ignore react-sticky-box@2.0.0 

https://github.com/codecks-io/react-sticky-box/issues/87

File: package.json [hunk 2]
```diff
@@ -254,6 +253,7 @@
     "progress": "^2.0.3",
     "qs": "^6.10.1",
     "rc-footer": "^0.6.8",
+    "rc-trigger": "^5.3.4",
```
Comment (afc163) [inline package.json:256]: 看上去这个 rc-trigger 也可以干掉了。

**Review by li-jia-nan — APPROVED** [review]

**Comment by li-jia-nan** [comment]
能 merge 了嘛？

**Comment by yoyo837** [comment]
> 能 merge 了嘛？

等等

File: package.json (hunk 2 above)
Comment (yoyo837) [reply package.json:256]: 好像不行哦, 还有俩组件没迁移到 `@rc-component/trigger`, 那俩组件有关 trigger 用例会不一致吧

Comment (yoyo837) [reply package.json:256]: 共享的 shared/demoTest

**Comment by danielberndt** [comment]
react-sticky-box 2.0.3 was just released to fix this. sorry for the hassle, esm is complicated 🙏

**Comment by li-jia-nan** [comment]
> react-sticky-box 2.0.3 was just released to fix this. sorry for the hassle, esm is complicated 🙏

thank you very much

**Pull request merged**